src/main/java/uk/co/hadoopathome/intellij/viewer/FileViewerToolWindow.java

```diff
@@ -168,22 +168,23 @@ protected Boolean doInBackground() {
             try {
               Reader reader =
                   currentFile.getName().endsWith("avro")
-                      ? new AvroReader(currentFile)
-                      : new ParquetReader(currentFile);
+                      ? new AvroFileReader(currentFile)
+                      : new ParquetFileReader(currentFile);
               List<String> records = reader.getRecords(numRecords);
               tableHandler.updateTable(records);
               dataRawTextArea.setText(StringUtils.join(records, "\n"));
               schemaTextPane.setText(reader.getSchema());
               fileInfoLabel.setText(
                   String.format("Displaying %d records from %s", records.size(), file.getPath()));
               return true;
-            } catch (OutOfMemoryError | IOException e) {
+            } catch (Throwable t) {
```

Comment:
Sadly necessary, as some internal Parquet/Avro reading errors can result in Throwables rather than Exceptions.